src/fastaudio/augment/spectrogram.py

```diff
@@ -5,11 +5,27 @@
 from fastcore.utils import ifnone
 from torch.nn import functional as F
 
-from ..core.spectrogram import AudioSpectrogram
+from ..core.spectrogram import AudioSpectrogram, AudioTensor
 from .signal import AudioPadType
 
 
-class CropTime(Transform):
+class InvalidEncodingData(Exception):
+    pass
+
+
+invalid_exception = InvalidEncodingData(
+    "You are trying to use a spectrogram transform on an audio tensor, check your pipeline"
+)
+
+
+class SpectrogramTransform(Transform):
+    "Helps prevent us trying to apply to Audio Tensors"
+
+    def encodes(self, audio: AudioTensor) -> AudioTensor:
+        raise InvalidEncodingData
```

Comment:
Raising a error is too hard and incompatible with the way fastai works in general. Suppose the user has a datablock with (AudioSpectrogram, AudioTensor) for some task he want to do, with this approach he can't use spectrogram augmentations. We could instead just show a warning, and return the AudioTensor so it remains compatible with fastai.